src/core/dataset.rs

```diff
@@ -11,6 +11,7 @@ pub trait Dataset<T: Number, U: Number>: std::fmt::Debug + Send + Sync {
     fn indices(&self) -> Vec<usize>;
     fn one_to_one(&self, left: usize, right: usize) -> U;
     fn query_to_one(&self, query: &[T], index: usize) -> U;
+    fn swap(&mut self, i: usize, j: usize);
```

Comment:
Add documentation to state the purpose of this method, and say that it is acceptable to panic if the indices are out of range.